src/other/metadata.jl

```diff
@@ -63,16 +77,19 @@ const COLMETADATA_EXAMPLE =
     """
 
 """
-    metadata(df::AbstractDataFrame, key::AbstractString; style::Bool=false)
-    metadata(dfr::DataFrameRow, key::AbstractString; style::Bool=false)
-    metadata(dfc::DataFrameColumns, key::AbstractString; style::Bool=false)
-    metadata(dfr::DataFrameRows, key::AbstractString; style::Bool=false)
+    metadata(df::AbstractDataFrame, key::AbstractString, [default]; style::Bool=false)
+    metadata(dfr::DataFrameRow, key::AbstractString, [default]; style::Bool=false)
+    metadata(dfc::DataFrameColumns, key::AbstractString, [default]; style::Bool=false)
+    metadata(dfr::DataFrameRows, key::AbstractString, [default]; style::Bool=false)
 
 Return table-level metadata value associated with `df` for key `key`.
 If `style=true` return a tuple of metadata value and metadata style.
 
 `SubDataFrame` and `DataFrameRow` expose only `:note`-style metadata of their
-parent.
+parent (except if `default` is passed and `key` does not exist).
```

Comment:
I don't think this precision is needed, as non-`:note`-style metadata is not exposed when `default` is provided either.
```suggestion
parent.
```